internal/controllers/mic_reconciler.go

```diff
@@ -193,23 +193,27 @@ func (mrhi *micReconcilerHelperImpl) updateStatusByMBSC(ctx context.Context, mic
 	}
 
 	patchFrom := client.MergeFrom(micObj.DeepCopy())
-	for _, imageState := range mbsc.Status.ImagesStates {
-		imageSpec := mrhi.micHelper.GetModuleImageSpec(micObj, imageState.Image)
-		if imageSpec == nil {
-			// image not found in spec, ignore
+	for _, mbscImageState := range mbsc.Status.Images {
+		micImageSpec := mrhi.micHelper.GetModuleImageSpec(micObj, mbscImageState.Image)
+		mbscImageSpec := mrhi.mbscHelper.GetImageSpec(mbsc, mbscImageState.Image)
+		if micImageSpec == nil || mbscImageSpec == nil || mbscImageSpec.Action != mbscImageState.Action {
+			// spec for image does not exists in either MIC or MBCS, which is not normal, or
+			// the action in MBSC status not equal to action in MBSC spec, which means that the status
+			// was not yet updated. In any case - ignore
```

Comment:
Isn't it possible that the image was removed in the module and it was removed from MIC spec and MBSC spec but still present in MIC status and MBSC status because the controllers didn't update it yet?

Reply:
it is , but it still means that we can ignore it